assets/js/googlesitekit/modules/datastore/modules.js

```diff
@@ -144,6 +144,27 @@ const fetchSetModuleActivationStore = createFetchStore( {
 	},
 } );
 
+const fetchCheckModuleAccessStore = createFetchStore( {
+	baseName: 'checkModuleAccess',
+	controlCallback: ( { slug } ) => {
+		return API.set( 'core', 'modules', 'check-access', { slug } );
+	},
+	reducerCallback: ( state, { access }, { slug } ) => {
+		return {
+			...state,
+			moduleAccess: {
+				[ slug ]: access,
+			},
```

Comment:
We need to destruct the existing `moduleAccess` object in the current state to make sure that existing information remains:

```suggestion
			moduleAccess: {
				...state.moduleAccess,
				[ slug ]: access,
			},
```